design/ceph/object/store.md

```diff
@@ -712,3 +712,84 @@ For external CephCephObjectStores (i.e., when `spec.gateway.externalRgwEndpoints
 vhost-style addressing should be configured on the host cluster, and `hosting.dnsNames` is
 irrelevant. The default `advertiseEndpoint` for external CephObjectStores is the first entry in the
 `spec.gateway.externalRgwEndpoints` list, which users should be able to override if desired.
+
+### Advanced user configuration
+
+RGW is a very active project with many existing features and regular new feature additions. Rook
+maintainers have a hard time catching up with RGW's latest features. In an effort to help users who
+want access to rare and new features, Rook can allow users to set arbitrary Ceph configs in their
+CephObjectStore CR. RGW config reference: https://docs.ceph.com/en/reef/radosgw/config-ref/
+
+This is an advanced feature. It allows users to modify many untested aspects of the RGW and object
+store. Any user configs that break the object store should be documented to be a responsibility of
+the user.
+
+```yaml
+spec:
+  metadataPool: # ...
+  dataPool: # ...
+  gateway:
+    rgwConfig:
+      conf-name: value
+      other_conf: value
+      debug_rgw: "20"
+    rgwCommandFlags:
+      flag-name: value
+      other_flag: value
+      rgw_enable_apis: s3, s3website, swift, swift_auth, sts, iam, notifications
+```
+
+CephCluster has a configuration section called `config` that sets mon config store values. `config`
+may be too vague in the context of the CephObjectStore CR. `rgwConfig` is a clearer naming for users
+and gives context that Ceph RGW configs are expected. A search engine search for "rgw config" also
+yields the correct Ceph documentation for users, which suggest the naming is apt.
+
+CephObjectStore will also add an `rgwCommandFlags` variable to allow users to set CLI flags to the
+`radosgw` deployment. Changes to these parameters will result in RGW daemon restarts, which can be
+desired behavior for config values like `rgw_enable_apis`. In testing, the `radosgw` command process
+supports adding multiple flags, and the last flag sequentially is used as the input source.
+
+Rook users can already use the `CephCluster.spec.config` section to control individual
+CephObjectStore configurations today. However, this requires users to know the name of the Ceph
+client user for each CephObjectStore. Adding this feature improves config override usability for
+object store users, and it keeps config overrides in the same place as the resource. This is a clear
+quality of life improvement for users managing object stores.
```

Comment:
Maybe the ordering issue raised in https://github.com/rook/rook/issues/14753 should also be mentioned here?  I think there are two problems which this feature would resolve.
1) knowing in advance what client name rook is going to use
2) the ordering issue in that trying apply config for a client name which doesn't exist yet will fail (and block re-provisioning a cluster)